libtenzir/builtins/operators/decapsulate.cpp

```diff
@@ -399,14 +409,29 @@ class plugin final : public operator_plugin<decapsulate_operator> {
     return {.transformation = true};
   }
 
+  auto make_function(invocation inv,
+                     session ctx) const -> failure_or<function_ptr> override {
+    auto expr = ast::expression{};
+    TRY(argument_parser2::function("tql2.decapsulate")
+          .add(expr, "<field>")
+          .parse(inv, ctx));
+    return function_use::make(
+      [expr = std::move(expr)](evaluator eval, session ctx) -> series {
+        auto series = eval(expr);
+        if (auto op = decapsulate(series, ctx.dh())) {
+          return op.value();
+        }
+        // TODO: Think about type
+        return series::null(null_type{}, series.length());
```

Comment:
If you know the exact that type that `decapsulate` produces, then you could use that here.

Reply:
Since the full type varies a bit depending on the packet, `null_type` should be fine here.

Reply:
I think in a huddle earlier, you also suggested `record_type{}`? Between both, which should we choose?

Reply:
That is also an option, but they currently behave different when it comes to warnings, because `record_type{}` basically claims that the resulting record would have no fields. `null_type` on the other hand just says "this could be anything". For example, we warn if we access an unknown field in a `record_type`, but not for `null_type`. So `null_type{}` is better than `record_type{}`. The other alternative is full-blown `record_type{...}` with all fields, but since this changes here depending on the content, that is not an option.

Reply:
I see. Thanks for the explanation!